pkg/network/usm/utils/file_registry.go

```diff
@@ -40,170 +81,160 @@ func NewFileRegistry() *FileRegistry {
 	)
 
 	return &FileRegistry{
+		procRoot:      util.GetProcRoot(),
 		byID:          make(map[PathIdentifier]*registration),
 		byPID:         make(map[uint32]pathIdentifierSet),
 		blocklistByID: make(pathIdentifierSet),
 		telemetry: registryTelemetry{
-			libHookFailed:               metricGroup.NewCounter("hook_failed"),
-			libRegistered:               metricGroup.NewCounter("registered"),
-			libAlreadyRegistered:        metricGroup.NewCounter("already_registered"),
-			libBlocked:                  metricGroup.NewCounter("blocked"),
-			libUnregistered:             metricGroup.NewCounter("unregistered"),
-			libUnregisterNoCB:           metricGroup.NewCounter("unregister_no_callback"),
-			libUnregisterErrors:         metricGroup.NewCounter("unregister_errors"),
-			libUnregisterFailedCB:       metricGroup.NewCounter("unregister_failed_cb"),
-			libUnregisterPathIDNotFound: metricGroup.NewCounter("unregister_path_id_not_found"),
+			fileHookFailed:               metricGroup.NewCounter("hook_failed"),
+			fileRegistered:               metricGroup.NewCounter("registered"),
+			fileAlreadyRegistered:        metricGroup.NewCounter("already_registered"),
+			fileBlocked:                  metricGroup.NewCounter("blocked"),
+			fileUnregistered:             metricGroup.NewCounter("unregistered"),
+			fileUnregisterNoCB:           metricGroup.NewCounter("unregister_no_callback"),
+			fileUnregisterErrors:         metricGroup.NewCounter("unregister_errors"),
+			fileUnregisterFailedCB:       metricGroup.NewCounter("unregister_failed_cb"),
+			fileUnregisterPathIDNotFound: metricGroup.NewCounter("unregister_path_id_not_found"),
 		},
 	}
 }
 
-// Register a ELF library root/libPath as be used by the pid
-// Only one registration will be done per ELF (system wide)
-func (r *FileRegistry) Register(root, libPath string, pid uint32, activate activationCB, deactivate deactivationCB) {
-	hostLibPath := root + libPath
-	pathID, err := NewPathIdentifier(hostLibPath)
+// Register inserts or updates a new file registration within to the `FileRegistry`;
+//
+// If no current registration exists for the given `PathIdentifier`, we execute
+// its *activation* callback. Otherwise, we increment the reference counter for
+// the existing registration if and only if `pid` is new;
+func (r *FileRegistry) Register(namespacedPath string, pid uint32, activationCB, deactivationCB callback) {
+	if activationCB == nil && deactivationCB == nil {
+		// if *both* callbacks are nil, there is no point in proceeding
+		return
+	}
```

Comment:
I think it is a rare and unwanted scenario, so we should add a log about it.

Reply:
furthermore, there is no point at registering if activeCB is nil.
if deactiveCB can be nil? TBH, I think not, but that's up to you.
thus you should change the code to be `if activationCB == nil {` or `if activationCB == nil || deactivationCB == nil {`

Reply:
to be fair I don't think there is a real use case where either would be nil, so I'm requiring both to be non-nil now